src/resource/solidDataset.test.ts

```diff
@@ -921,6 +925,92 @@ describe("saveSolidDatasetAt", () => {
         deletions: [],
       });
     });
+
+    it("returns a meaningful error when the server returns a 403", async () => {
+      const mockFetch = jest
+        .fn(window.fetch)
+        .mockReturnValue(
+          Promise.resolve(new Response("Not allowed", { status: 403 }))
+        );
+
+      const mockDataset = getMockUpdatedDataset(
+        {
+          additions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some.vocab/subject"),
+              DataFactory.namedNode("https://some.vocab/predicate"),
+              DataFactory.namedNode("https://some.vocab/object"),
+              undefined
+            ),
+          ],
+          deletions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some-other.vocab/subject"),
+              DataFactory.namedNode("https://some-other.vocab/predicate"),
+              DataFactory.namedNode("https://some-other.vocab/object"),
+              undefined
+            ),
+          ],
+        },
+        "https://some.pod/resource"
+      );
+
+      const fetchPromise = saveSolidDatasetAt(
+        "https://some.pod/resource",
+        mockDataset,
+        {
+          fetch: mockFetch,
+        }
+      );
+
+      await expect(fetchPromise).rejects.toThrow(
+        "Storing the Resource at `https://some.pod/resource` failed: 403 Forbidden.\n\n" +
+          "The changes that were sent to the Pod are listed below.\n\n"
+      );
+    });
+
+    it("returns a meaningful error when the server returns a 404", async () => {
+      const mockFetch = jest
+        .fn(window.fetch)
+        .mockReturnValue(
+          Promise.resolve(new Response("Not found", { status: 404 }))
+        );
+
+      const mockDataset = getMockUpdatedDataset(
+        {
+          additions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some.vocab/subject"),
+              DataFactory.namedNode("https://some.vocab/predicate"),
+              DataFactory.namedNode("https://some.vocab/object"),
+              undefined
+            ),
+          ],
+          deletions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some-other.vocab/subject"),
+              DataFactory.namedNode("https://some-other.vocab/predicate"),
+              DataFactory.namedNode("https://some-other.vocab/object"),
+              undefined
+            ),
+          ],
+        },
+        "https://some.pod/resource"
+      );
+
+      const fetchPromise = saveSolidDatasetAt(
+        "https://some.pod/resource",
+        mockDataset,
+        {
+          fetch: mockFetch,
+        }
+      );
+
+      await expect(fetchPromise).rejects.toThrow(
+        "Storing the Resource at `https://some.pod/resource` failed: 404 Not Found.\n\n" +
+          "The changes that were sent to the Pod are listed below.\n\n"
```

Comment:
The expected values for the changelog are not tested for. Should they be ?

Reply:
Yep, that's the case for every test that no longer explicitly tests on `Error` but just on the error message. My reasoning here was that for these tests, the actual contents of the dataset isn't too relevant, as long as it says the status code, URL to be saved on, and that it will display the changes. (The actual changes dataset is empty.)

Whether the dataset's Markdown representation is correct is already covered by another test, and this way, we don't need to update this specific test if we change the human-readable representation of a dataset. That said, if we somehow remove the call to `solidDatasetAsMarkdown()` in the code, this test will not catch that, so... Pros and cons, I guess. My deliberation landed on this approach, but happy to change to the other if you thing those pros/cons outweigh this one's.

Reply:
No, that seems perfectly reasonable. We can always test for the content of the message in an end-to-end test if the need arises, but I agree that having separate tests for the markdownification of a dataset on one hand, and for the contextual info on the other feels like the right call.